### PR TITLE
Miscellaneous cleanup/refactoring in `resolve` and `syntax::ext`

### PR DESCRIPTION
**Pull request by jseyfried**
r? @nrc

### CODE REVIEW

File: src/librustc/hir/mod.rs [hunk 1]
```diff
@@ -475,8 +475,6 @@ pub struct MacroDef {
     pub attrs: HirVec<Attribute>,
     pub id: NodeId,
     pub span: Span,
-    pub imported_from: Option<Name>,
```
Comment (jseyfried) [inline src/librustc/hir/mod.rs]: This was always `None` due to #37213.

File: src/libsyntax/ast.rs [hunk 2]
```diff
@@ -160,7 +155,13 @@ pub struct PathSegment {
     /// this is more than just simple syntactic sugar; the use of
     /// parens affects the region binding rules, so we preserve the
     /// distinction.
-    pub parameters: PathParameters,
+    pub parameters: Option<P<PathParameters>>,
```
Comment (petrochenkov) [inline src/libsyntax/ast.rs]: Does this introduce distinction between `segment` and `segment::<>` or is it purely a size optimization?

Comment (jseyfried) [inline src/libsyntax/ast.rs]: I amended so that it is purely a size optimization.

Comment (petrochenkov) [inline src/libsyntax/ast.rs]: Could you add a comment telling that this is a size optimization and what `None` means?

Comment (jseyfried) [inline src/libsyntax/ast.rs]: Will do.

**Comment by nrc** [comment]
r=me with the comment @petrochenkov requested

**Comment by jseyfried** [comment]
This merged in #38499.

**Pull request closed without merging**